src/uu/split/src/filenames.rs

```diff
@@ -2,528 +2,181 @@
+/// assert_eq!(it.next().unwrap(), "chunk_01.txt");
+/// assert_eq!(it.next().unwrap(), "chunk_02.txt");
 /// ```
-pub struct FilenameFactory {
-    additional_suffix: String,
-    prefix: String,
-    suffix_length: usize,
-    use_numeric_suffix: bool,
-}
-
-impl FilenameFactory {
-    /// Create a new instance of this struct.
-    ///
-    /// For an explanation of the parameters, see the struct documentation.
+impl FilenameIterator {
     pub fn new(
         prefix: String,
         additional_suffix: String,
         suffix_length: usize,
         use_numeric_suffix: bool,
-    ) -> FilenameFactory {
-        FilenameFactory {
+    ) -> FilenameIterator {
+        let radix = if use_numeric_suffix { 10 } else { 26 };
+        let number = if suffix_length == 0 {
+            Number::DynamicWidth(DynamicWidthNumber::new(radix))
+        } else {
+            Number::FixedWidth(FixedWidthNumber::new(radix, suffix_length))
+        };
+        FilenameIterator {
             prefix,
             additional_suffix,
-            suffix_length,
-            use_numeric_suffix,
+            number,
+            first_iteration: true,
         }
     }
+}
+
+impl Iterator for FilenameIterator {
+    type Item = String;
 
-    /// Construct the filename for the specified element of the output collection of files.
-    ///
-    /// For an explanation of the parameters, see the struct documentation.
-    ///
-    /// If `suffix_length` has been set to a positive integer and `i`
-    /// is greater than or equal to the number of strings that can be
-    /// represented within that length, then this returns `None`. For
-    /// example:
-    ///
-    /// ```rust,ignore
-    /// use crate::filenames::FilenameFactory;
-    ///
-    /// let prefix = String::new();
-    /// let suffix = String::new();
-    /// let width = 1;
-    /// let use_numeric_suffix = true;
-    /// let factory = FilenameFactory::new(prefix, suffix, width, use_numeric_suffix);
-    ///
-    /// assert_eq!(factory.make(10), None);
-    /// ```
-    pub fn make(&self, i: usize) -> Option<String> {
+    fn next(&mut self) -> Option<Self::Item> {
+        // The first and third parts are just taken directly from the
+        // struct parameters unchanged.
         let prefix = self.prefix.clone();
-        let suffix1 = match (self.use_numeric_suffix, self.suffix_length) {
-            (true, 0) => Some(num_prefix(i)),
-            (false, 0) => str_prefix(i),
-            (true, width) => num_prefix_fixed_width(i, width),
-            (false, width) => str_prefix_fixed_width(i, width),
-        }?;
-        let suffix2 = &self.additional_suffix;
-        Some(prefix + &suffix1 + suffix2)
+        let suffix = &self.additional_suffix;
+        if self.first_iteration {
+            self.first_iteration = false;
+        } else {
+            self.number.increment().ok()?;
+        }
+        Some(format!("{}{}{}", prefix, self.number, suffix))
```

Comment:
The `FilenameIterator` repeatedly increments a counter stored in `self.number` and formats it into a string with a specified prefix and suffix.